ballerina/natives.bal

```diff
@@ -0,0 +1,383 @@
+    'class: "io.ballerina.stdlib.observe.nativeimpl.AddTagToMetrics",
+    name: "addTagToMetrics"
+} external;
+
+# Finish the current span.
+#
+# + spanId - Id of span to finish
+# + return - An error if an error occurred while finishing the span
+public function finishSpan(int spanId) returns error? = @java:Method {
+    'class: "io.ballerina.stdlib.observe.nativeimpl.FinishSpan",
+    name: "finishSpan"
+} external;
+
+# Retrieve a map of span context data.
+#
+# + return - Map of span context data (traceId and spanId).
+public isolated function getSpanContext() returns map<string> = @java:Method {
+    'class: "io.ballerina.stdlib.observe.nativeimpl.GetSpanContext",
+    name: "getSpanContext"
+} external;
+
+# Retrieve all registered metrics including default metrics from the ballerina runtime, and user defined metrics.
+#
+# + return - Array of all registered metrics
+public function getAllMetrics() returns Metric[] = @java:Method {
+    'class: "io.ballerina.stdlib.observe.nativeimpl.GetAllMetrics",
+    name: "getAllMetrics"
+} external;
+
+# Retrieve the specific metric that is described by the given name and tags.
+#
+# + name - Name of the metric to lookup
+# + tags - The key/value pair tags associated with the metric that should be looked up
+# + return - The metric instance
+public function lookupMetric(string name, map<string>? tags = ()) returns Counter|Gauge? = @java:Method {
+    'class: "io.ballerina.stdlib.observe.nativeimpl.LookupMetric",
+    name: "lookupMetric"
+} external;
+
+# This represents the metric type - counter, that can be only increased by an integer number.
+#
+# + name - Name of the counter metric
+# + description - Description of the counter metric
+# + metricTags - Tags associated with the counter metric
+public  class Counter {
+
+    public string name;
+    public string description;
+    public map<string> metricTags;
+
+    # This instantiates the Counter object. Name field is mandatory, and description and tags fields
+    # are optional and have its own default values when no params are passed.
+    #
+    # + name - Name of the Counter instance
+    # + desc - Description of the Counter instance. If no description is provided, the the default empty string
+    #          will be used
+    # + tags - The key/value pair of Tags. If no tags are provided, the default nil value will be used
+    public function init(string name, string? desc = "", map<string>? tags = ()) {
+        self.name = name;
+        if (desc is string) {
+            self.description = desc;
+        } else {
+            self.description = "";
+        }
+        if (tags is map<string>) {
+            self.metricTags = tags;
+        } else {
+            self.metricTags = DEFAULT_TAGS;
+        }
+        externCounterInit(self);
+    }
+
+    //# Performs the necessary native operations during the initialization of the counter.
+    //function initialize() = external;
```

Comment:
Let's remove this. (seems to be an unused part from long time back.)

Reply:
Removed in https://github.com/ballerina-platform/module-ballerina-observe/pull/1/commits/ed8df9f5f8ca5121b0dd0a3a15cb43f5ed96749a